### PR TITLE
GHA: Run unit tests in CI

### PR DESCRIPTION
**Pull request by limbonaut**
Run unit tests in CI using Windows and Linux runners. This will help us detect compilation and runtime issues early on.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (bruno-garcia) [inline .github/workflows/ci.yml]:
```suggestion
      - "main"
```

File: .github/workflows/unit_tests.yml
Comment (bruno-garcia) [inline .github/workflows/unit_tests.yml:22]: Why the additional step instead of this?

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:22]: Can't specify which submodules to check out... We don't need `sentry-native` and such, which also packs its own submodules on top. If you prefer simpler, I can remove that step.

File: .github/workflows/ci.yml
Comment (limbonaut) [reply .github/workflows/ci.yml]: That was [temporary](https://github.com/getsentry/sentry-godot/pull/37#issuecomment-2520008953).
Dropped.

File: .github/workflows/unit_tests.yml
Comment (bruno-garcia) [reply .github/workflows/unit_tests.yml:22]: All good

**Comment by limbonaut** [comment]
@bitsandfoxes @bruno-garcia
Good to merge? It's still waiting for approval.

**Review by bitsandfoxes — APPROVED** [review]
Awesome stuff!

File: .github/workflows/unit_tests.yml
Comment (bitsandfoxes) [inline .github/workflows/unit_tests.yml:75]: I assume this step fails if the tests fail?

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:75]: Strangely, docs don't specify. I'll test.

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:75]: ~~I'm not sure about `timeout-minutes` - docs don't specify if it fails the step. Need to check it.~~ (Works as expected).
According to docs, the process should return:
```
0   = all tests success
100 = ends with test failures
101 = ends with test warnings
```

**Pull request merged**
